hyperhtml.js

```diff
@@ -193,321 +156,692 @@ var hyperHTML = (function () {'use strict';
     };
   }
 
-  // update attributes node
-  //
-  // render`<a href="${url}" onclick="${click}">${name}</a>`;
-  //
-  // Note: attributes with a special meaning like DOM Level 0
-  //       listeners or accessors properties are directly set
-  function setAttribute(node, attribute) {
+  // `<div class="${'attr'}"></div>`
+  function setAttribute(attribute, removeAttributes) {
     var
       name = attribute.name,
+      node = attribute.ownerElement,
       isEvent = name.slice(0, 2) === 'on',
       isSpecial = name in node && !SHOULD_USE_ATTRIBUTE.test(name),
       oldValue
     ;
-    if (isSpecial) node.removeAttribute(name);
+    if (isSpecial) removeAttributes.push(node, name);
     return isSpecial ?
       function specialAttr(newValue) {
         if (oldValue !== newValue) {
+          if (isEvent && oldValue && ('handleEvent' in oldValue)) {
+            handleEvent(node, 'remove', name, oldValue);
+          }
           oldValue = newValue;
-          if (isEvent) {
-            node[name] = 'handleEvent' in newValue ?
-              newValue.handleEvent.bind(newValue) :
-              newValue;
+          if (isEvent && newValue && ('handleEvent' in newValue)) {
+            handleEvent(node, 'add', name, newValue);
           } else {
             node[name] = newValue;
           }
         }
       } :
       function attr(newValue) {
         if (oldValue !== newValue) {
```

Comment:
actually, after changing it, I've realized I don't have any empty return anywhere else ... leaving as it is for now :stuck_out_tongue: